R/pipeline-helpers.R

```diff
@@ -167,3 +167,63 @@ unprotect_integer_cols <- function(d) {
   names(d) <- sub('^X([0-9]+)$', '\\1', names(d))
   d
 }
+
+
+#' gdp_deflator
+#'
+#' Calculate a gross domestic product (GDP) implicit price deflator between two years.
+#'
+#' @param year Year for which to calculate deflator
+#' @param base_year Base year in deflator calculation
+#' @details The GDP deflator is a measure of price inflation/deflation with respect to a specific base year.
+#' @note This returns slightly different values if \code{OLD_DATA_SYSTEM_BEHAVIOR} is TRUE.
+#' @return Deflator (GDP in \code{year} divided by GDP in \code{base_year}).
+#' @source U.S. Bureau of Economic Analysis, Gross domestic product (implicit price deflator) [A191RD3A086NBEA], retrieved from FRED, Federal Reserve Bank of St. Louis; https://fred.stlouisfed.org/series/A191RD3A086NBEA, April 12, 2017
+#' @author BBL
+#' @export
+#' @examples
+#' gdp_deflator(2010, base_year = 1990)
+gdp_deflator <- function(year, base_year) {
+
+  if(OLD_DATA_SYSTEM_BEHAVIOR) {
+    if(year == 2010 & base_year == 1990) return(1.510)   # conv_1990_2010_USD
+    if(year == 2007 & base_year == 1990) return(1.470)   # conv_1990_2007_USD
+    if(year == 2005 & base_year == 1990) return(1.383)   # conv_1990_2005_USD
+    if(year == 1975 & base_year == 1990) return(0.4649)  # conv_1990_1975_USD
+    if(year == 1975 & base_year == 1996) return(0.4049)  # conv_1996_1975_USD
+    if(year == 1975 & base_year == 1997) return(0.3983)  # conv_1997_1975_USD
+    if(year == 1975 & base_year == 1998) return(0.3939)  # conv_1998_1975_USD
+    if(year == 1975 & base_year == 1999) return(0.3883)  # conv_1999_1975_USD
+    if(year == 1975 & base_year == 2000) return(0.380)   # conv_2000_1975_USD
+    if(year == 1975 & base_year == 2001) return(0.3711)  # conv_2001_1975_USD
+    if(year == 1975 & base_year == 2002) return(0.3647)  # conv_2002_1975_USD
+    if(year == 1975 & base_year == 2003) return(0.3571)  # conv_2003_1975_USD
+    if(year == 1975 & base_year == 2004) return(0.3472)  # conv_2004_1975_USD
+    if(year == 1975 & base_year == 2005) return(0.3362)  # conv_2005_1975_USD
+    if(year == 1975 & base_year == 2006) return(0.3257)  # conv_2006_1975_USD
+    if(year == 1975 & base_year == 2007) return(0.317)   # conv_2007_1975_USD
+    if(year == 1975 & base_year == 2008) return(0.3104)  # conv_2008_1975_USD
+    if(year == 1975 & base_year == 2009) return(0.3104)  # conv_2009_1975_USD
+  }
+
+  # This time series is the BEA "A191RD3A086NBEA" product
+  # Downloaded April 13, 2017 from https://fred.stlouisfed.org/series/A191RD3A086NBEA
+  gdp_years <- 1929:2016
+  gdp <- c(9.896, 9.535, 8.555, 7.553, 7.345, 7.749, 7.908, 8.001, 8.347,
```

Comment:
this may confuse people. this value is actually (nominal_gdp)/(real_gdp)*100. a reader may think this is actual GDP.
i would suggest changing to gdp_deflator, but then that name is already in use. so maybe "implicit_price_index"?